### PR TITLE
Identity README additions for 1.1.0 release

### LINKED ISSUE #12372: [FEATURE REQ] AzureCliCredential and all new credentials need to be added to docs
**Library or service name.**
Azure Identity
 
**Is your feature request related to a problem? Please describe.**
Cannot find docs on how to use AzureCliCredential or any of the new credentials, i.e. VSCodeCredential, etc

azsdke2e

### PR DESCRIPTION
**Pull request by jianghaolu**
resolves #12372

### CODE REVIEW

**Comment by JimSuplizio** [comment]
@jianghaolu, keyvault is failing its From Source runs because spring-cloud-azure-appconfiguration-config, has a dependency on azure-security-keyvault-secrets. I've got a [PR ](https://github.com/Azure/azure-sdk-for-java/pull/13647) open to fix this. You'll have to rebase when I get it merged.

**Comment by jianghaolu** [comment]
Thanks! I wasn't able to figure that out. Glad you took a look at my pr

File: sdk/identity/azure-identity/README.md [hunk 1]
```diff
@@ -159,7 +118,14 @@ If the environment configuration is not present or incomplete, the `DefaultAzure
 require platform support. See the
 [managed identity documentation](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/services-support-managed-identities) for more details on this.
 
-If a managed identity isn't available, and the application is running on Windows, the `DefaultAzureCredential` will then attempt reading from a local user token cache. `AZURE_USERNAME` environment variable must be specified if there are more than one accounts in the cache. The local token cache is shared between this library, Visual Studio (2019+), and Azure CLI. See [Enable applications for shared token cache credential](#enable-applications-for-shared-token-cache-credential) to see how to populate / clean up this token cache.
+If a managed identity isn't available, the `DefaultAzureCredential` will then attempt reading the signed in user from a set of local developer tools, in the following order
```
Comment (g2vinay) [inline sdk/identity/azure-identity/README.md:121]: Should we also mention about EnvironmentCredential not being available too  and then attempting below chain of creds ?

File: sdk/identity/azure-identity/README.md [hunk 2]
```diff
@@ -274,14 +250,39 @@ public void createUserNamePasswordCredential() {
...
+See more about how to configure an AAD application for interactive browser authentication and listen on a port locally in [Enable applications for interactive browser oauth 2 flow](#enable-applications-for-interactive-browser-oauth-2-flow)
+
+<!-- embedme ../../keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/IdentityReadmeSamples.java#L190-L203 -->
+```java
+/**
+ * Authenticate interactively in the browser.
+ */
+public void createInteractiveBrowserCredential() {
+    InteractiveBrowserCredential interactiveBrowserCredential = new InteractiveBrowserCredentialBuilder()
+        .port(8765)
```
Comment (g2vinay) [inline sdk/identity/azure-identity/README.md:265]: Client Id is required for Interactive Browser Cred, so this won't work as it is.

File: sdk/identity/azure-identity/README.md [hunk 3]
```diff
@@ -300,10 +301,76 @@ public void createAuthCodeCredential() {
...
+See more about how to configure your IntelliJ IDEA in [Sign in Azure Toolkit for IntelliJ for IntelliJCredential](#sign-in-azure-toolkit-for-intellij-for-intellijcredential).
+
+<!-- embedme ../../keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/IdentityReadmeSamples.java#L174-L188 -->
+```java
+/**
+ * Authenticate with IntelliJ IDEA.
+ */
+public void createIntelliJCredential() {
+    IntelliJCredential intelliJCredential = new IntelliJCredentialBuilder()
+        // KeePass configuration required for Windows
```
Comment (g2vinay) [inline sdk/identity/azure-identity/README.md:337]: Maybe elaborate the comment here ?
required only for Windows.
No configuration needed for Linux / Mac

File: sdk/identity/azure-identity/README.md [hunk 4]
```diff
@@ -334,6 +401,129 @@ public void createChainedCredential() {
...
+The Visual Studio Code authentication is handled by an integration with the Azure Account extension. To use, install the Azure Account extension, then use View->Command Palette to execute the “Azure: Sign In” command:
+
+![vscode sign in](./images/vscode-sign-in.png)
+
+This will open a browser that allows you to sign in to Azure. Once you have completed the login process, you can close the browser as directed. Running your application (either in the debugger or anywhere on the development machine) will use the credential from your sign-in.
+
+![vscode logged in](./images/vscode-logged-in.png)
+
+### Configure `DefaultAzureCredential`
+
```
Comment (g2vinay) [inline sdk/identity/azure-identity/README.md:518]: Should this be first in the order ?

File: sdk/keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/IdentityReadmeSamples.java [hunk 5]
```diff
@@ -137,4 +145,60 @@ public void createChainedCredential() {
...
+            .credential(intelliJCredential)
+            .buildClient();
+    }
+
+    /**
+     * Authenticate interactively in the browser.
+     */
+    public void createInteractiveBrowserCredential() {
+        InteractiveBrowserCredential interactiveBrowserCredential = new InteractiveBrowserCredentialBuilder()
+            .port(8765)
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/IdentityReadmeSamples.java:195]: client id too, needs to be added.

File: sdk/identity/azure-identity/README.md [hunk 6]
```diff
@@ -300,10 +301,76 @@ public void createAuthCodeCredential() {
...
+        .vaultUrl("https://{YOUR_VAULT_NAME}.vault.azure.net")
+        .credential(intelliJCredential)
+        .buildClient();
+}
+```
+
+### Authenticating a user account with Visual Studio Code
+This example demonstrates authenticating the `KeyClient` from the [azure-security-keyvault-keys][keys_client_library] client library using the `VisualStudioCodeCredential` on a workstation with Visual Studio Code installed, and the user has signed in with an Azure account.
+
+See more about how to configure your Visual Studio Code in [Sign in Visual Studio Code for VisualStudioCodeCredential](#sign-in-visual-studio-code-for-visualstudiocodecredential)
```
Comment (g2vinay) [inline sdk/identity/azure-identity/README.md:352]: Sign in Visual Studio Code for...  ---->.  Sign in Visual Studio Code Azure Extension for....

File: sdk/identity/azure-identity/README.md (hunk 1 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/README.md:121]: It's mentioned in line 117 - just not shown in GitHub diff

File: sdk/identity/azure-identity/README.md (hunk 2 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/README.md:266]: added

File: sdk/identity/azure-identity/README.md (hunk 3 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/README.md]: modified

File: sdk/identity/azure-identity/README.md (hunk 4 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/README.md]: Moved

File: sdk/identity/azure-identity/README.md (hunk 6 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/README.md]: Modified

File: sdk/keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/IdentityReadmeSamples.java (hunk 5 above)
Comment (jianghaolu) [reply sdk/keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/IdentityReadmeSamples.java:196]: Added

**Pull request merged**